### PR TITLE
Run dark mode test only on mojave

### PR DESCRIPTION
**Pull request by simonhong**
Fix https://github.com/brave/brave-browser/issues/4073

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Windows
  - [ ] macOS
  - [ ] Linux
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests && npm run test-security`) on
  - [ ] Windows
  - [x] macOS
  - [ ] Linux
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [x] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone

## Test Plan:
`yarn test brave_browser_tests --filter=BraveDarkModeObserverTest.*`

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

### CODE REVIEW

**Comment by bsclifton** [comment]
Test security failed on these; rebased so that CI would run again

**Comment by simonhong** [comment]
@bsclifton all green! :)

**Comment by simonhong** [comment]
kindly ping..

File: browser/ui/brave_dark_mode_observer_browsertest_mac.mm [hunk 1]
```diff
@@ -27,23 +27,25 @@ void SetBraveThemeType(Profile* profile, BraveThemeType type) {
...
-  SetBraveThemeType(profile, BraveThemeType::BRAVE_THEME_TYPE_LIGHT);
-  EXPECT_EQ(
-      ui::NativeTheme::GetInstanceForNativeUi(),
-      BraveDarkModeObserver::current_native_theme_for_testing_);
-
-  SetBraveThemeType(profile, BraveThemeType::BRAVE_THEME_TYPE_DARK);
-  EXPECT_EQ(
-      ui::NativeThemeDarkAura::instance(),
-      BraveDarkModeObserver::current_native_theme_for_testing_);
+  if (@available(macOS 10.14, *)) {
```
Comment (bsclifton) [inline browser/ui/brave_dark_mode_observer_browsertest_mac.mm]: this is pretty interesting syntax; do you have more info about it?

**Comment by bsclifton** [comment]
After merging, can you please create PRs for 0.64.x and 0.63.x for this too? Should be very low risk- easy win to include 😄

File: browser/ui/brave_dark_mode_observer_browsertest_mac.mm (hunk 1 above)
Comment (simonhong) [reply browser/ui/brave_dark_mode_observer_browsertest_mac.mm]: for more detail - https://clang.llvm.org/docs/LanguageExtensions.html#objective-c-available

**Pull request merged**